rust/vote-tx-v2/src/gen_tx/vote.rs

```diff
@@ -0,0 +1,77 @@
+//! A generalized tx vote struct.
+
+use minicbor::{Decode, Decoder, Encode};
+
+use crate::{encoded_cbor::EncodedCbor, Cbor};
+
+/// `Vote` array struct length
+const VOTE_LEN: u64 = 3;
```

Comment:
It is defined from the cddl spec 
```cddl
vote<choice-t, proof-t, prop-id-t>  = [
    choices<choice-t>,
    proof<proof-t>,
    prop-id<prop-id-t>,
]
```
so basically, `vote` type here a cbor encoded array with the 3 field in it.
Also I've discussed with @stevenj that we should allow only definite sized arrays and maps in such cases (because cbor also allows to have indefinite size arrays and maps and cddl does not specify it). I will put a note into our spec about it.

Reply:
@cong-or so probably we will also need to revisit ledger encoding implementation to double check that it encodes as a finite cbor arrays and cbor maps for the types, but its not urgent and for now its totally ok if its not strictly verifies it.